pkgs/development/python-modules/python-snappy/default.nix

```
@@ -0,0 +1,29 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, cffi
+, snappy
+}:
+
+buildPythonPackage rec {
+  pname = "python-snappy";
+  version = "0.5.3";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "8a7f803f06083d4106d55387d2daa32c12b5e376c3616b0e2da8b8a87a27d74a";
+  };
+
+  buildInputs = [ cffi snappy ];
```

```suggestion
  buildInputs = [ cffi snappy ];
```
Shouldn't cffi be propagated? Have a look at https://github.com/NixOS/nixpkgs/pull/56027 for a better version of this expression.

---

Yes #56027 is a better version. How should I handle this? I certainly appreciate fastparquet being added!

---

Just use the commit from there and we'll rebase which ever commit gets merged second.
Btw, do you have any idea about that `ModuleNotFoundError` problem?

---

Fixed. I just copied the file in #56027 for python-snappy

---

I'm also wondering why this `except` doesn't trigger: https://github.com/andrix/python-snappy/blob/0.5.3/snappy/snappy.py#L50

---

I found a solution: a simple `rm -r snappy` before the tests does the trick because it makes `test_snappy.py` import the installed package instead of `./snappy`. Not sure if there's a way to prevent local imports.

---

`from __future__ import absolute_import` inside the test script does not do the trick sadly. Reminder to myself: learn more about Python's `import`.
Maybe @FRidh has some insight into this?

---

python by default looks in the current directory first for a package before trying to import from site-packages. Good simple solution I like `rm -r snappy`. I have pushed this fix.